module/VuFind/src/VuFind/Search/Base/Options.php

```diff
@@ -434,7 +434,17 @@ public function __construct(\VuFind\Config\PluginManager $configLoader)
         $this->loadResultsWithJs = (bool)($searchSettings->General->load_results_with_js ?? true);
         $this->topPaginatorStyle = $searchSettings->General->top_paginator
             ?? ($this->loadResultsWithJs ? 'simple' : false);
-        $this->hiddenSortOptions = $searchSettings?->HiddenSorting?->pattern?->toArray() ?? [];
+
+        // Merge hidden sort options and any labels into a single array:
+        $hiddenSortOptions = $searchSettings?->HiddenSorting?->pattern?->toArray() ?? [];
+        $hiddenSortOptionLabels = $searchSettings?->HiddenSorting?->label?->toArray() ?? [];
+        foreach ($hiddenSortOptions as $key => $pattern) {
+            $label = (string)($hiddenSortOptionLabels[$key] ?? $key);
+            $this->hiddenSortOptions[] = [
+                'label' => ctype_digit($label) ? null : $label,
+                'pattern' => $pattern,
+            ];
+        }
```

Comment:
Is it worth refactoring this to an `initializeHiddenSortOptions` support method, since there's quite a bit of logic in play? Might make the constructor feel a little cleaner.

Reply:
Yes, done.